### ISSUE TITLE
Working with the return of decltype

### ISSUE THREAD

**Issue by linuswagner** [usethesource/clair#62]
Apparently, we don't support working with the return on `decltype`.
The code
```
void f()
{
        decltype(someExpression)::g(1);
}
```
gives (`notGood.cpp` contains the code above):
```
DecltypeSpecifier: decltype(someExpression)
|lib://clair/src/lang/cpp/M3.rsc|(9983,280,<176,0>,<177,243>): Java("RuntimeException","NYI at |file:///home/linus/Documents/mapper/src/main/rascal/notGood.cpp|(20,24)")
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<4255,0>,<4255,0>))
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTDecltypeSpecifier.accept(|unknown:///CPPASTDecltypeSpecifier.java|(0,0,<74,0>,<74,0>))
        at lang.cpp.internal.Parser.lambda$visit$9(|unknown:///Parser.java|(0,0,<925,0>,<925,0>))
        at java.util.Spliterators$ArraySpliterator.forEachRemaining(|unknown:///Spliterators.java|(0,0,<948,0>,<948,0>))
        at java.util.stream.ReferencePipeline$Head.forEach(|unknown:///ReferencePipeline.java|(0,0,<658,0>,<658,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<924,0>,<924,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<891,0>,<891,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<858,0>,<858,0>))
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTQualifiedName.accept(|unknown:///CPPASTQualifiedName.java|(0,0,<236,0>,<236,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<3301,0>,<3301,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<2624,0>,<2624,0>))
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTIdExpression.accept(|unknown:///CPPASTIdExpression.java|(0,0,<90,0>,<90,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<3283,0>,<3283,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<2622,0>,<2622,0>))
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTFunctionCallExpression.accept(|unknown:///CPPASTFunctionCallExpression.java|(0,0,<208,0>,<208,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<3752,0>,<3752,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<3558,0>,<3558,0>))
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTExpressionStatement.accept(|unknown:///CPPASTExpressionStatement.java|(0,0,<68,0>,<68,0>))
        at lang.cpp.internal.Parser.lambda$visit$47(|unknown:///Parser.java|(0,0,<3697,0>,<3697,0>))
        at java.util.Spliterators$ArraySpliterator.forEachRemaining(|unknown:///Spliterators.java|(0,0,<948,0>,<948,0>))
        at java.util.stream.ReferencePipeline$Head.forEach(|unknown:///ReferencePipeline.java|(0,0,<658,0>,<658,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<3696,0>,<3696,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<3548,0>,<3548,0>))
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTCompoundStatement.accept(|unknown:///CPPASTCompoundStatement.java|(0,0,<88,0>,<88,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<1234,0>,<1234,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<965,0>,<965,0>))
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTFunctionDefinition.accept(|unknown:///CPPASTFunctionDefinition.java|(0,0,<191,0>,<191,0>))
        at lang.cpp.internal.Parser.visit(|unknown:///Parser.java|(0,0,<836,0>,<836,0>))
        at org.eclipse.cdt.internal.core.dom.parser.ASTTranslationUnit.accept(|unknown:///ASTTranslationUnit.java|(0,0,<278,0>,<278,0>))
        at lang.cpp.internal.Parser.convertCdtToRascal(|unknown:///Parser.java|(0,0,<628,0>,<628,0>))
        at lang.cpp.internal.Parser.parseCppToM3AndAst(|unknown:///Parser.java|(0,0,<476,0>,<476,0>))
        at parseCppToM3AndAst(|lib://clair/src/lang/cpp/M3.rsc|(10256,5,<177,236>,<177,241>))
        at main(|file:///home/linus/Documents/mapper/src/main/rascal/MApper/UIXML_CPP_Rule.rsc|(719,15,<31,4>,<31,19>))
        at $shell$(|prompt:///|(0,6,<1,0>,<1,6>)WARNING: ResourcesPlugin was null, can't get workspace; not overriding include files
ok
```

What works though is:
```
void f()
{
        decltype(someExpression);
}
```

**Comment by jurgenvinju** [comment]
Thanks @linuswagner ; will have a look after the weekend.

**Comment by jurgenvinju** [comment]
This was fixed an released with version 0.11.0
